test/client_test.exs

```diff
@@ -354,4 +355,93 @@ defmodule Sentry.ClientTest do
              Sentry.capture_message("something happened", extra: %{metadata: [keyword: "list"]})
            end) =~ "Failed to send Sentry event. Unable to encode JSON"
   end
+
+  describe "exceptions, throws, and exits" do
+    setup :verify_on_exit!
+
+    setup do
+      modify_env(:sentry,
+        dsn: "http://public:secret@localhost:0/1",
+        client: Sentry.MockClient
+      )
+
+      :ok
+    end
+
+    test "handles exits" do
+      expect(Sentry.MockClient, :post, fn _url, _headers, _body -> exit(:timeout) end)
+
+      try do
+        apply(Event, :not_a_function, [])
```

Comment:
To simplify this test, could you use `Sentry.capture_message/2` to avoid the `try/catch`?

Reply:
I'll try using `Sentry.capture_message/2`, not sure why I chose `try/catch` back then.

Reply:
@ruslandoga awesome. Lmk when this is ready for review. 👍

Reply:
@whatyouhide should be ready :)